cmd/probe/internal/component/mysql/wesql_manager.go

```diff
@@ -0,0 +1,329 @@
+		return false
+	}
+	if db == nil {
+		return false
+	}
+
+	defer db.Close()
+	var leaderRecord RowMap
+	sql := "select * from information_schema.wesql_cluster_global;"
+	err = QueryRowsMap(db, sql, func(rMap RowMap) error {
+		if rMap.GetString("ROLE") == "Leader" {
+			leaderRecord = rMap
+		}
+		return nil
+	})
+	if err != nil {
+		mgr.Logger.Errorf("error executing %s: %v", sql, err)
+		return false
+	}
+
+	if len(leaderRecord) > 0 {
+		return true
+	}
+	return false
+}
+
+// IsClusterInitialized is a method to check if cluster is initailized or not
+func (mgr *WesqlManager) IsClusterInitialized(ctx context.Context, cluster *dcs.Cluster) (bool, error) {
+	clusterInfo := mgr.GetClusterInfo(ctx, nil)
+	if clusterInfo != "" {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (mgr *WesqlManager) GetClusterInfo(ctx context.Context, cluster *dcs.Cluster) string {
+	var db *sql.DB
+	var err error
+	if cluster != nil {
+		db, err = mgr.GetLeaderConn(ctx, cluster)
+		if err != nil {
+			mgr.Logger.Infof("Get leader conn failed: %v", err)
+			return ""
+		}
+		if db != nil {
+			defer db.Close()
+		}
+	} else {
+		db = mgr.DB
+
+	}
+	var clusterID, clusterInfo string
+	err = db.QueryRowContext(ctx, "select cluster_id, cluster_info from mysql.consensus_info").
+		Scan(&clusterID, &clusterInfo)
+	if err != nil {
+		mgr.Logger.Error("Cluster info query failed: %v", err)
+	}
+	return clusterInfo
+}
+
+func (mgr *WesqlManager) Promote(ctx context.Context, cluster *dcs.Cluster) error {
+	isLeader, _ := mgr.IsLeader(ctx, nil)
+	if isLeader {
+		return nil
+	}
+
+	db, err := mgr.GetLeaderConn(ctx, cluster)
+	if err != nil {
+		return errors.Wrap(err, "Get leader conn failed")
+	}
+	if db != nil {
+		defer db.Close()
+	}
+
+	currentMember := cluster.GetMemberWithName(mgr.GetCurrentMemberName())
+	addr := cluster.GetMemberAddr(*currentMember)
+	resp, err := db.Exec(fmt.Sprintf("call dbms_consensus.change_leader('%s:13306');", addr))
+	if err != nil {
+		mgr.Logger.Errorf("promote err: %v", err)
+		return err
+	}
+
+	mgr.Logger.Infof("promote success, resp:%v", resp)
+	return nil
+}
+
+func (mgr *WesqlManager) IsPromoted(ctx context.Context) bool {
+	isLeader, _ := mgr.IsLeader(ctx, nil)
+	return isLeader
+}
+
+func (mgr *WesqlManager) Demote(context.Context) error {
+	return nil
+}
+
+func (mgr *WesqlManager) Follow(ctx context.Context, cluster *dcs.Cluster) error {
+	return nil
+}
+
+func (mgr *WesqlManager) GetHealthiestMember(cluster *dcs.Cluster, candidate string) *dcs.Member {
+	return nil
+}
+
+func (mgr *WesqlManager) HasOtherHealthyLeader(ctx context.Context, cluster *dcs.Cluster) *dcs.Member {
+	clusterLocalInfo, err := mgr.GetClusterLocalInfo(ctx)
+	if err != nil || clusterLocalInfo == nil {
+		mgr.Logger.Errorf("Get cluster local info failed: %v", err)
+		return nil
+	}
+
+	if clusterLocalInfo.GetString("ROLE") == "Leader" {
+		// I am the leader, just return nil
+		return nil
+	}
+
+	leaderAddr := clusterLocalInfo.GetString("CURRENT_LEADER")
```

Comment:
using the constant string instead of determined string, include `Leader`, check them all